scipy/sparse/linalg/_eigen/_svds.py

```diff
@@ -341,46 +316,24 @@ def matmat_XH_X(x):
         eigvals, eigvec = eigsh(XH_X, k=k, tol=tol ** 2, maxiter=maxiter,
                                 ncv=ncv, which=which, v0=v0)
 
-    # Gramian matrices have real non-negative eigenvalues.
-    eigvals = np.maximum(eigvals.real, 0)
-
-    # Use the sophisticated detection of small eigenvalues from pinvh.
-    t = eigvec.dtype.char.lower()
-    factor = {'f': 1E3, 'd': 1E6}
-    cond = factor[t] * np.finfo(t).eps
-    cutoff = cond * np.max(eigvals)
-
-    # Get a mask indicating which eigenpairs are not degenerately tiny,
-    # and create the re-ordered array of thresholded singular values.
-    above_cutoff = (eigvals > cutoff)
-    nlarge = above_cutoff.sum()
-    nsmall = k - nlarge
-    slarge = np.sqrt(eigvals[above_cutoff])
-    s = np.zeros_like(eigvals)
-    s[:nlarge] = slarge
+    u = X_matmat(eigvec)
     if not return_singular_vectors:
-        return np.sort(s)
-
-    if n > m:
-        vlarge = eigvec[:, above_cutoff]
-        ularge = (X_matmat(vlarge) / slarge
-                  if return_singular_vectors != 'vh' else None)
-        vhlarge = _herm(vlarge)
+        s = svd(u, compute_uv=False)
+        return s[::-1]
+
+    # compute the left singular vectors of X and update the right ones
+    # accordingly
+    u, s, vh = svd(u, full_matrices=False)
+    u = u[:, ::-1]
+    s = s[::-1]
+    vh = vh[::-1]
+    jobu = return_singular_vectors in {True, 'u'}
+    jobv = return_singular_vectors in {True, 'vh'}
+    if not transpose:
+        u = u if jobu else None
+        vh = vh @ _herm(eigvec) if jobv else None
     else:
-        ularge = eigvec[:, above_cutoff]
-        vhlarge = (_herm(X_matmat(ularge) / slarge)
-                   if return_singular_vectors != 'u' else None)
-
-    u = (_augmented_orthonormal_cols(ularge, nsmall, random_state)
-         if ularge is not None else None)
-    vh = (_augmented_orthonormal_rows(vhlarge, nsmall, random_state)
-          if vhlarge is not None else None)
-
-    indexes_sorted = np.argsort(s)
-    s = s[indexes_sorted]
-    if u is not None:
-        u = u[:, indexes_sorted]
-    if vh is not None:
-        vh = vh[indexes_sorted]
-
-    return u, s, vh
+        u_tmp = eigvec @ _herm(vh) if jobu else None
+        vh = _herm(u) if jobv else None
+        u = u_tmp
+    return u, s, vh
```

Comment:
Add a new line at the end of the file

Reply:
Yes. I'll commit this:
```suggestion
    return u, s, vh

```
to fix ```scipy/sparse/linalg/_eigen/_svds.py:339:20: W292 no newline at end of file``` shortly.